.github/workflows/cicd.yaml

```diff
@@ -337,32 +351,50 @@ jobs:
 
   upgrade-test:
     runs-on: ubuntu-latest
-    needs: [build]
+    needs: [build,context]
+    permissions:
+      packages: read
+    env:
+      IMAGE: ${{ needs.context.outputs.build_image }}
     steps:
       - uses: actions/checkout@v3
+      - name: Login with ghcr
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
       - name: Install yq and bash
         run: |
           sudo snap install yq
           sudo apt update
           sudo apt install bash -y
         shell: bash
-      - uses: actions/download-artifact@v3
-        with:
-          name: images
-      - name: Load Docker images
-        run: |
-          docker load -i ${GITHUB_SHA}_image.tar
-        shell: bash
       - uses: ./.github/actions/k8s-version-config
         name: Setup k8s cluster
         with:
           k8s-version: v1.25
+      - name: Configure Cluster
+        run: |
+          kubectl create ns connaisseur
+          kubectl create secret generic ${imagepullsecret} \
+            --from-file=.dockerconfigjson=$HOME/.docker/config.json \
+            --type=kubernetes.io/dockerconfigjson \
+            --namespace=connaisseur
       - uses: actions/checkout@v3
         with:
           ref: "master"
+      - name: Checkout required current files
+        uses: Bhacaz/checkout-files@v2
+        with:
+          files: tests/integration/ghcr-values.yaml
+          branch: ${{ github.head_ref || github.ref_name }}
       - name: Configure Connaisseur (master)
         run: |
-          yq e '.detectionMode=true' -i helm/values.yaml
+          envsubst < tests/integration/ghcr-values.yaml > update
+          yq '. *+ load("update")' -i helm/values.yaml
```

Comment:
Not knowing what this does, I'd assume its similar to below. If so, why different syntax. If not, maybe a comment would be nice

Reply:
utlimately, it overwrites/extends the other yaml with another yaml provided via 'load'. The other style of writing still exists throughout our code at numerous places and below I essentially just copy-pasted (I believe). Want to clean that up in a general refactoring after this MR

Reply:
I would not think this is worthy of a comment

Reply:
I don't think I fully understand the comment. What do you mean by "It's really different"? You mean it looks different or it creates different output?
Ultimately, the former version appears to use `*` at its core and I use `*+`. Mine extends, the former overwrites arrays. My plan was to harmonize the writings in the next round when I really get to the `integration-test.sh`

Reply:
If `yq '. *+ load("update")' -i helm/values.yaml` and `yq eval-all --inplace 'select(fileIndex == 0) * select(fileIndex == 1)' helm/values.yaml update` do the same, why use both and not just one of those consistently?

Reply:
What I meant is that `yq '. *+ load("update")' -i helm/values.yaml` has a way different yq syntax from `yq eval-all --inplace 'select(fileIndex == 0) * select(fileIndex == 1)'`  and seems to do a similar job, in which case we shouldn't use different syntax.
But if yours updates and extends and the select syntax only updates, then there's a good reason to use "your" syntax instead of the old one (though if it's just more powerful we should also switch the other's over (at some point))

Reply:
will be in the next refactor